runtime/v2/shim/publisher.go

```diff
@@ -43,12 +44,22 @@ type item struct {
 
 // NewPublisher creates a new remote events publisher
 func NewPublisher(address string) (*RemoteEventsPublisher, error) {
+	return newPublisher(context.Background(), address)
+}
+
+// NewPublisherWithContext creates a new remote events publisher with context
+func NewPublisherWithContext(ctx context.Context, address string) (*RemoteEventsPublisher, error) {
+	return newPublisher(ctx, address)
+}
+
+func newPublisher(ctx context.Context, address string) (*RemoteEventsPublisher, error) {
 	client, err := ttrpcutil.NewClient(address)
 	if err != nil {
 		return nil, err
 	}
 
 	l := &RemoteEventsPublisher{
+		context: ctx,
```

Comment:
Carrying context as a struct field is a [bad tone](https://go.dev/blog/context-and-structs),
if we want to preserve those log fields, I'd suggest adding log field instead:

```go
l := &RemoteEventsPublisher{
    logger: log.GetLogger(ctx),
...
}
```

Reply:
Thank you for the tip.

I passed the context to `l. processQueue` and modified the context field of the task service.
PTAL